jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -148,23 +132,20 @@ public BaseImageWithAuthorization call()
 
         // If failed, then, retrieve base registry credentials and try with retrieved credentials.
         // TODO: Refactor the logic in RetrieveRegistryCredentialsStep out to
-        // registry.credentials.RegistryCredentialsRetriever to avoid this direct executor hack.
-        ListeningExecutorService directExecutorService = MoreExecutors.newDirectExecutorService();
-        RetrieveRegistryCredentialsStep retrieveBaseRegistryCredentialsStep =
+        // registry.credentials.RegistryCredentialsRetriever.
+        Credential registryCredential =
             RetrieveRegistryCredentialsStep.forBaseImage(
-                directExecutorService,
-                buildConfiguration,
-                progressEventDispatcher.newChildProducer());
+                    buildConfiguration, progressEventDispatcher.newChildProducer())
+                .call();
```

Comment:
Note to reviewers: now we can directly run ".call()", without futzing with an executor service and `NonBlockingSteps.get()`.

Reply:
This call of `RetrieveRegistryCredentialsStep` seems out of place. Should it be an actual step in our `buildSteps` in the same way `retrieveTargetRegistryCredentials` generates a step in our build?

Reply:
This is related to my comment below about renaming these classes (and potentially convert `call()` to a static method). For example, if regard this "Step" as a task and assume it had a method
```
getCredentials(buildConfiguration, ...);
```
instead of
```
forBaseImage(buildConfiguration, ...).call();
```
this wouldn't feel weird. As I said in the other comment, we should definitely think about renaming these classes and methods.

> Should it be an actual step in our

For this, I think this is OK. The "steps" enqueued `StepsRunner` are the unit of works that we think worth parallelization. At least I am not changing the current behavior in that the `PullBaseImageStep` was getting credentials serially (using a direct executor). That looks OK, because getting a credential is fast enough, all happening locally (e.g., reading `~/.docker/config.json`, ...).